packages/workflow/src/RoutingNode.ts

```diff
@@ -623,27 +624,28 @@ export class RoutingNode {
 			);
 		}
 
-		const executePaginationFunctions = {
-			...executeSingleFunctions,
-			makeRoutingRequest: async (requestOptions: DeclarativeRestApiSettings.ResultOptions) => {
-				return await this.rawRoutingRequest(
-					executeSingleFunctions,
-					requestOptions,
-					credentialType,
-					credentialsDecrypted,
-				).then(
-					async (data) =>
-						await this.postProcessResponseData(
-							executeSingleFunctions,
-							data,
-							requestData,
-							itemIndex,
-							runIndex,
-						),
-				);
-			},
+		const makeRoutingRequest = async (requestOptions: DeclarativeRestApiSettings.ResultOptions) => {
+			return await this.rawRoutingRequest(
+				executeSingleFunctions,
+				requestOptions,
+				credentialType,
+				credentialsDecrypted,
+			).then(
+				async (data) =>
+					await this.postProcessResponseData(
+						executeSingleFunctions,
+						data,
+						requestData,
+						itemIndex,
+						runIndex,
+					),
+			);
 		};
 
+		const executePaginationFunctions = Object.create(executeSingleFunctions, {
+			makeRoutingRequest: { value: makeRoutingRequest },
+		}) as IExecutePaginationFunctions;
```

Comment:
I think this could be a child class of `ExecuteSingleContext`  in the future

Reply:
agreed. There is already a [ticket](https://linear.app/n8n/issue/CAT-313/) that will cover this, and bunch more code around `RoutingNode`.